docs/paper/dev/misc/debugging.md

```diff
@@ -0,0 +1,76 @@
+# Debugging your plugin
+
+Debugging your plugin is vital to being able to fix bugs and issues in your plugin. This page will cover some of the most common debugging techniques.
+
+### Randomly printing to the console
+
+One of the most common debugging techniques is to print to the console. This is likely something you've done before as its very simple. 
+It has a few downsides though. It can be hard to find the print statements in the console, and it can be hard to remove them all when you're done debugging.
+
+When debugging, you can use `System.out.println("");` to print to the console. It is recommended to use your plugin's logger instead though, 
+as it will be easier to know which plugin the log has come from. This can be done simply with:
+
+```java
+plugin.getLogger().warning("SuperDuperBad Thing has happened");
+```
+
+:::note[Logger Levels]
+
+In some consoles, using the `warning` level will print the message in different colours.
+This can be useful for finding your print statements in the console.
+
+:::
+
+### Using a remote debugger
+
+A debugger is a tool that allows you to pause your code at a certain point and inspect the values of variables.
+This can be very useful for finding out why your code isn't working as expected and also for finding out where your code is going wrong.
+
+#### Setting up the debugger
+
+To use a debugger, you need to set up your IDE to use it. This is different for each IDE, but for the sake of this guide, we will be using IntelliJ IDEA.
+
+To set up a debugger in IntelliJ, you need to create a new run configuration.
+You can do this by clicking the dropdown next to the run button and clicking `Edit Configurations...`:
+
+![](./assets/config_dropdown.png)
+
+Then, click the `+` button in the top left and select `Remote JVM Debug`. You can then name the configuration whatever you want, and click `Apply`:
+
+![](./assets/config_add.png)
+
+Finally, copy the command line arguments from the window, and paste these into your server's startup script. Once you have done this, you can click `OK`.
+Once your server is running, you can use the bug icon in the top right to connect your debugger to the server:
+
+![](./assets/debugger_connect.png)
+
```

Comment:
I feel like in addition to just remote debugging (maybe only exclusively), we should include direct debugging? I rarely use remote debugging, only in specific situations where its required. Usually now, I just use jmp's run-server gradle plugin and run `runServer` gradle task with debug which automatically starts the debugger.

Reply:
Most people won't be testing their plugin in a vacuum/inside rundev but some shared test server, so remote debugging is still the more important section imo. But adding something for direct debugging would still be nice

Reply:
Added a note about this at the bottom. I'd rather not add too much documentation here, as it will be somewhat redundant when there is the project README